### DISCUSSION TITLE
uWSGI - Socket File Missing

### DISCUSSION THREAD

**Discussion by CuriousCorvid** [searxng/searxng#3831] [Q&A] [answered]
Hello, 
I'm trying to get SearXNG to work on Apache for a long time now. I manage to access my site when I use the webapp.py. I manage to proxy it to my own domain. But I don't manage to run it deamonized yet, using uWSGI.

The following line in the `searxng.ini` file gives me trouble
`socket = /usr/local/searxng/run/socket`.
This file(?) does not exist. It's nowhere mentioned what it contains. Apparently the installation script just installs it, but since I do it manually, I don't know what to do. Could anyone help?

**Comment by return42** [comment, marked as answer]
Its a UNIX socket, generated by the uWSGI application.  A [UNIX socket](https://en.wikipedia.org/wiki/Unix_domain_socket), AKA Unix Domain Socket, is an inter-process communication mechanism that allows bidirectional data exchange between processes running on the same machine.

The uWSGI (python) process listen on this socket and the HTTP server (proxy) is connected to this socket.

Related comments:

- https://github.com/searxng/searxng/discussions/3764#discussioncomment-10443520
- https://github.com/searxng/searxng/discussions/3052#discussioncomment-7790896

**Reply by CuriousCorvid** [reply]
Rather than the socket, I have now set 
`http = 127.0.0.1:8888`
which I found here: https://searx.github.io/searx/admin/installation-uwsgi.html#searx-uwsgi
Is there any disadvantage to this?

If someone could just share all relevant folders for an Apache and Nginx installation on Github, that would make things a lot easier.

**Reply by virtadpt** [reply]
There are no disadvantages, and it makes things a bit easier in the long run.  As for Nginx, here's the relevant part of my Nginx configs:

```
server {
    server_name  searxng.example.com;

    access_log /var/log/nginx/searxng.example.com.access.log;
    error_log /var/log/nginx/searxng.example.com.error.log;

    root /home/virtadpt/searxng/searx;
    location /static { }

    location / {
        proxy_pass http://127.0.0.1:8888;
        proxy_set_header Host $host;
        proxy_set_header Connection $http_connection;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Scheme $scheme;
        proxy_buffering off;

        # Let's up the timeouts on SearxNG.
        proxy_connect_timeout   600;
        proxy_send_timeout      600;
        proxy_read_timeout      600;
        send_timeout            600;
    }
}
```

**Reply by CuriousCorvid** [reply]
Thank you, I appreciate it. Then I will leave the config on `http://127.0.0.1:8888;`.
